src/content/app/species/components/species-main-view/prepareStatistics.ts

```diff
@@ -0,0 +1,235 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  getStatsForSection,
+  speciesStatsSectionNames,
+  type SpeciesStatsSection,
+  type StatsSection
+} from '../../state/general/speciesGeneralHelper';
+
+import type { SpeciesStatistics } from 'src/content/app/species/state/api/speciesApiTypes';
+import type { ExampleFocusObject } from 'src/shared/state/genome/genomeTypes';
+
+/**
+ * RULES FOR THE CLIENT FOR DISPLAYING STATISTICS IN THE EXPANDABLE SECTIONS
+ *
+ * - For assembly_stats: anything can be null except for chromosomes.
+ *   If chromosomes are null, all fields in the section should be null.
+ * - For coding_stats: anything can be null except for coding_genes.
+ *   If coding_genes is null, all fields in the section should be null
+ * - For non_coding_stats: anything can be null except for non_coding_genes.
+ *   If non_coding_genes is null, all fields in the section should be null
+ * - For pseudogene_stats, anything can be null except for pseudogenes.
+ *   If pseudogenes field is null, all fields in the section should be null
+ * - For variation_stats, anything can be null except for short_variants.
+ *   If short_variants is null, all fields in the section should be null
+ * - For homology_stats, both coverage and coverage_explanation
+ *   should at the same time be either null or non-null
+ * - For regulation_stats, anything can be null
+ */
+const curateSpeciesStatistics = (
+  speciesStats: SpeciesStatistics
+): SpeciesStatistics => {
+  return {
+    assembly_stats: curateAssemblyStatistics(speciesStats.assembly_stats),
+    coding_stats: curateCodingGeneStatistics(speciesStats.coding_stats),
+    non_coding_stats: curateNonCodingGeneStatistics(
+      speciesStats.non_coding_stats
+    ),
+    pseudogene_stats: curatePseudogeneStatistics(speciesStats.pseudogene_stats),
```

Comment:
Why not make all of the methods above end with `Stats` rather than `Statistics` or vice verse for consistency?

Reply:
✅